### PR TITLE
Add GitLab OAuth provider documentation

### PR DESCRIPTION
**Pull request by fh1ch**
Documents the GitLab OAuth integration. This feature is an open PR at openshift/origin#6923.

### CODE REVIEW

File: install_config/configuring_authentication.adoc [hunk 1]
```diff
@@ -875,6 +875,55 @@ application]. The application must be configured with a callback URL of
...
+oauthConfig:
+  ...
+  identityProviders:
+  - name: gitlab <1>
+    challenge: false <2>
+    login: true <3>
+    mappingMethod: claim <4>
+    provider:
+      apiVersion: v1
+      kind: GitLabIdentityProvider
```
Comment (liggitt) [inline install_config/configuring_authentication.adoc]: add the `ca` field and description

File: install_config/configuring_authentication.adoc [hunk 2]
```diff
@@ -875,6 +875,55 @@ application]. The application must be configured with a callback URL of
 <6> The client secret issued by GitHub.
 ====
 
+[[GitLab]]
+
+=== GitLab
+
+Set *GitLabIdentityProvider* in the `*identityProviders*` stanza to use
+https://gitlab.com/[GitLab.com] or any other GitLab instance as an identity provider, using the
+http://doc.gitlab.com/ce/integration/oauth_provider.html[OAuth integration].
```
Comment (liggitt) [inline install_config/configuring_authentication.adoc]: Add gitlab version requirement info

**Comment by fh1ch** [comment]
Required GitLab version and `ca` field including description added.

**Comment by adellape** [comment]
:+1: Thanks.

**revhistory note:**

(Install & Config -> Configuring Authentication)

```
Added a section for the new link:../install_config/configuring_authentication.html#GitLab[GitLab identity provider].
```

**Pull request merged**